sonar-perl-plugin/src/main/java/com/github/otrosien/sonar/perl/tap/TestHarnessJUnitReader.java

```diff
@@ -0,0 +1,175 @@
+package com.github.otrosien.sonar.perl.tap;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import com.github.otrosien.sonar.perl.tap.TestHarnessReport.Test;
+import com.github.otrosien.sonar.perl.tap.TestHarnessReport.TestDetail;
+import com.github.otrosien.sonar.perl.tap.TestHarnessReport.TestDetail.TestDetailBuilder;
+import com.github.otrosien.sonar.perl.tap.TestHarnessReport.TestHarnessReportBuilder;
+
+/*
+A difficulty with JUnit reports is that we need to recover test file path
+info.
+
+There are mainly two Perl modules that supports generating JUnit XML,
+one is TAP::Harness::JUnit, the other is TAP::Formatter::JUnit.
+
+For TAP::Formatter::JUnit, we would support if it dumps to individual
+files under a directory via setting PERL_TEST_HARNESS_DUMP_TAP.  The junit
+xml files are named by appending .junit.xml to the t/xxx.t form.
+
+For TAP::Harness::JUnit, we would assume its namemangle is "none" or
+"perl".  We cannot support "hudson" mode as we can't safely recover test
+file path in this case.  Actually even "perl" mode is not 100% safe
```

Comment:
Can you elaborate in which circumstances "perl" mode fails? If it's something the user should be aware of, please add it to the README.